drools-wb-screens/drools-wb-enum-editor/drools-wb-enum-editor-client/src/test/java/org/drools/workbench/screens/enums/client/editor/EnumEditorPresenterTests.java

```diff
@@ -0,0 +1,217 @@
+
+    @Mock
+    private DefaultFileNameValidator mockFileNameValidator;
+
+    @Mock
+    private EnumService enumService;
+    private Caller<EnumService> enumServiceCaller;
+
+    @Mock
+    private ObservablePath path;
+
+    @Mock
+    private PlaceRequest place;
+
+    @Mock
+    private Overview overview;
+
+    @GwtMock
+    private ViewDRLSourceWidget sourceWidget;
+
+    @Captor
+    private ArgumentCaptor<List<EnumRow>> enumsArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> enumStringArgumentCaptor;
+
+    private Event<NotificationEvent> mockNotification = new EventSourceMock<NotificationEvent>() {
+        @Override
+        public void fire( final NotificationEvent event ) {
+            //Do nothing. Default implementation throws a RuntimeException
+        }
+    };
+
+    private EnumResourceType type;
+
+    private EnumEditorPresenter presenter;
+
+    private EnumModelContent content;
+    private EnumModel model;
+
+    @Before
+    public void setup() {
+        //Mock EnumResourceType
+        this.type = GWT.create( EnumResourceType.class );
+
+        //Mock FileMenuBuilder usage since we cannot use FileMenuBuilderImpl either
+        when( mockFileMenuBuilder.addSave( any( MenuItem.class ) ) ).thenReturn( mockFileMenuBuilder );
+        when( mockFileMenuBuilder.addCopy( any( ObservablePath.class ),
+                                           eq( mockFileNameValidator ) ) ).thenReturn( mockFileMenuBuilder );
+        when( mockFileMenuBuilder.addRename( any( ObservablePath.class ),
+                                             eq( mockFileNameValidator ) ) ).thenReturn( mockFileMenuBuilder );
+        when( mockFileMenuBuilder.addDelete( any( ObservablePath.class ) ) ).thenReturn( mockFileMenuBuilder );
+        when( mockFileMenuBuilder.addValidate( any( Command.class ) ) ).thenReturn( mockFileMenuBuilder );
+        when( mockFileMenuBuilder.addNewTopLevelMenu( any( MenuItem.class ) ) ).thenReturn( mockFileMenuBuilder );
+
+        when( mockVersionRecordManager.getCurrentPath() ).thenReturn( path );
+        when( mockVersionRecordManager.getPathToLatest() ).thenReturn( path );
+
+        this.model = new EnumModel( "'Fact.field' : ['a', 'b']" );
+        this.content = new EnumModelContent( model,
+                                             overview );
+
+        when( enumService.loadContent( path ) ).thenReturn( content );
+
+        when( view.getContent() ).thenReturn( new ArrayList<EnumRow>() {{
+            add( new EnumRow( "Fact",
+                              "field",
+                              "['a', 'b']" ) );
+        }} );
+
+        this.enumServiceCaller = new CallerMock<EnumService>( enumService );
+
+        this.presenter = new EnumEditorPresenter( view,
+                                                  enumServiceCaller,
+                                                  type ) {
+            {
+                //Yuck, yuck, yuck... the class hierarchy is really a mess
```

Comment:
@Rikkola I agree (this is the second time I've had to set ```@Injected'ed``` fields in a static initializer block. Ensuring ```BaseEditor``` and ```KieEditor``` use Constructor injection would help for sure.. as we'd need to pass objects through parent constructors. This would fix another issue I found where ```BaseEditor``` and ```KieEditor``` both ```@Inject Event<Notification>``` leading to the super and subclasses both having instances of the event with the same field name... which screwed up tests (I've removed the duplicate in a previous PR).